resolve/index.go

```diff
@@ -229,6 +248,19 @@ func (ix *RuleIndex) FindRulesByImport(imp ImportSpec, lang string) []FindResult
 	return results
 }
 
+// FindRulesByImportCrslv attempts to resolve an import to a rule first by
+// checking the rule index, then if no matches are found any registered
+// CrossResolve implementations are called.
+func (ix *RuleIndex) FindRulesByImportCrslv(c *config.Config, rc *repo.RemoteCache, imp ImportSpec, lang string) []FindResult {
```

Comment:
Let's call this `FindRulesByImportWithConfig` instead. It shouldn't need `rc` if that's dropped from `CrossResolve`.

Let's add `// Deprecated: use FindRulesByImportWithConfig instead` to the doc comment for `FindRulesByImport`. It's unfortunate that we can't consolidate them, but the `*config.Config` seems necessary for this to work.